src/command/defaults/ClearCommand.php

```diff
@@ -0,0 +1,178 @@
+
+use InvalidArgumentException;
+use pocketmine\command\CommandSender;
+use pocketmine\command\utils\InvalidCommandSyntaxException;
+use pocketmine\item\LegacyStringToItemParser;
+use pocketmine\lang\TranslationContainer;
+use pocketmine\player\Player;
+use pocketmine\utils\TextFormat;
+use function array_merge;
+use function count;
+use function is_numeric;
+
+class ClearCommand extends VanillaCommand{
+
+	public function __construct(string $name){
+		parent::__construct(
+			$name,
+			"%pocketmine.command.clear.description",
+			"%pocketmine.command.clear.usage"
+		);
+		$this->setPermission("pocketmine.command.clear.self;pocketmine.command.clear.other");
+	}
+
+	public function execute(CommandSender $sender, string $commandLabel, array $args){
+		if(!$this->testPermission($sender)){
+			return true;
+		}
+
+		if(count($args) > 3){
+			throw new InvalidCommandSyntaxException();
+		}
+
+		$target = null;
+		if($sender instanceof Player){
+			if(!$sender->hasPermission("pocketmine.command.clear.self")){
+				$sender->sendMessage(new TranslationContainer(TextFormat::RED . "%commands.generic.permission"));
+				return true;
+			}
+
+			$target = $sender;
+		}else{
+			if(count($args) < 1){
+				throw new InvalidCommandSyntaxException();
+			}
+		}
+
+		if(isset($args[0])){
+			if(!$sender->hasPermission("pocketmine.command.clear.other")){
+				$sender->sendMessage(new TranslationContainer(TextFormat::RED . "%commands.generic.permission"));
+				return true;
+			}
+
+			$target = $sender->getServer()->getPlayer($args[0]);
+		}
+
+		if($target === null){
+			$sender->sendMessage(new TranslationContainer(TextFormat::RED . "%commands.generic.player.notFound"));
+			return true;
+		}
+
+		$item = null;
+		if(isset($args[1])){
+			try{
+				$item = LegacyStringToItemParser::getInstance()->parse($args[1]);
+			}catch(InvalidArgumentException $e){
+				//vanilla checks this at argument parsing layer, can't come up with a better alternative
+				$sender->sendMessage(new TranslationContainer(TextFormat::RED . "%commands.give.item.notFound", [$args[1]]));
+				return true;
+			}
+		}
+
+		$maxCount = -1;
+		if(isset($args[2])){
+			if(!is_numeric($args[2])){
+				$maxCount = -1;
+			}
+
+			if($item !== null){
+				$item->setCount($maxCount = $this->getInteger($sender, $args[2], 0));
+			}
+		}
+
+		//checking players inventory for all the items matching the criteria
+		if($item !== null and $item->getMeta() === 0 and $maxCount === 0){
+			$count = 0;
+			$contents = array_merge($target->getInventory()->all($item), $target->getArmorInventory()->all($item));
+			foreach($contents as $content){
+				$count += $content->getCount();
+			}
+
+			if($count > 0){
+				$sender->sendMessage(new TranslationContainer("%commands.clear.testing", [$target->getName(), $count]));
+			}else{
+				$sender->sendMessage(new TranslationContainer(TextFormat::RED . "%commands.clear.failure.no.items", [$target->getName()]));
+			}
+
+			return true;
+		}
```

Comment:
I don't understand the purpose of this entire block. The point of `/clear` is to remove items.

Reply:
Discussed in above comment.